apps/dashboard/src/utils/beforeLoadUtil.ts

```diff
@@ -0,0 +1,11 @@
+import { useSettingsStore } from '@/stores/settings.store';
+import { populateStoresFromToken } from '@sudosos/sudosos-frontend-common';
+import apiService from '@/services/ApiService';
+
+export default async function beforeLoad() {
+
+  const settingsStore = useSettingsStore();
+  await settingsStore.fetchMaintenanceMode();
+
+  await populateStoresFromToken(apiService);
```

Comment:
And here catch the possible error from `populateStoresFromToken` and log out if need:
```js
clearTokenInStorage();
const authStore = useAuthStore();
authStore.logout();
```